array_linked_list.py/arrays.py

```diff
@@ -3,54 +3,71 @@
 # Do not use any of the built in array functions for this exercise
 class array:
     def __init__(self, capacity):
-        # Your code here
         self.capacity = capacity
         # cheating a little bit here by using a python list; however, won't use any built in array functions
         self.elements = [None] * capacity
-        pass
+        self.count = 0
+        
+        
 
 
-# Double the size of the given array
-def resize_array():
+# Double the size of the given array (which is what most languages will end up doing)
+def resize_array(array):
     # Your code here
-    pass
+    new_capacity = array.capacity * 2
+    new_elements = [None] * new_capacity
+    # for every item in previous array, set value in new_elements equal to item in previous array
+    for i in range(array.capacity):
+        new_elements[i] = array.elements[i]
+    # set array equal to new elements and new capacity
+    array.elements = new_elements
+    array.capacity = new_capacity
 
 
 # Return an element of a given array at a given index
-def array_read():
+def array_read(array, index):
     # Throw an error if array is out of the current count
     # Your code here
-    pass
+    if index >= array.count:
+        print("Error! Index " + str(index) +" out of range")
+        return None
+    # again, cheating here because we're using a list as the structure to store
+    # last time that we'll cheat!
+    return array.elements[index]
+    
 
 
 # Insert an element in a given array at a given index
-def array_insert():
+# need index of where you're inserting
+def array_insert(array, value, index):
```

Comment:
The built-in insert method for lists in Python 3 allows you to insert items at an index greater than or equal to the length of the list. If the index is greater than or equal to the length of the list, the item will automatically be appended to the end of the list.

The example below will print out [1, 2, 3, 4] when executed in Python: 

```
l = [1, 2, 3]
l.insert(100, 4)
print(l)
```

Also, the built-in insert method for lists in Python 3 allows you to insert items at an index less than the negative conversion of the length of the list. If the index is less than the negative conversion of the length of the list, the item will automatically be added to the beginning of the list.

The example below will print out [0, 2, 3, 4] when executed in Python: 

```
l = [1, 2, 3]
l.insert(-700, 0)
print(l)
```

Lastly, your code should also account for negative index insertions.